### PR TITLE
PPCDP-1711: add CDP's audience log membership endpoints

### PR DESCRIPTION
**Pull request by havaker**

### CODE REVIEW

**Comment by havaker** [comment]
@kororokke I've changed the base branch as suggested.

File: customer_data_platform/authorized_api/customer_data_platform_log_api.yaml [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+          $ref: "../components.yaml#/components/responses/UnprocessableEntityError"
+        500:
+          $ref: "../components.yaml#/components/responses/InternalServerError"
+  /api/cdp/log/v1/app/{app_id}/audience/{audience_id}/version/{audience_version}/profile/last:
+    get:
+      summary: List last profiles belonging to given audience
+      security:
+        - JWTAuth: []
+      operationId: get-log-audience-version-profile-last
+      description: Returns at most 1000 items
```
Comment (prk3) [inline customer_data_platform/authorized_api/customer_data_platform_log_api.yaml:61]: While technically true, this sentence might suggest that you can't get more than 1k profiles. Let's clarify this.
```suggestion
      description: By default, returns the last 100 items. With the use of limit and offest parameters, it is possible to extract all profiles belonging to the audience.
```

File: customer_data_platform/authorized_api/customer_data_platform_log_api.yaml [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+      parameters:
+        - $ref: "../components.yaml#/components/parameters/AppId"
+        - in: path
+          name: audience_id
+          description: An audience UUID for which profiles are requested
```
Comment (prk3) [inline customer_data_platform/authorized_api/customer_data_platform_log_api.yaml:66]: These explanations barely sound like English. Let's improve them a little before making the docs public.
```suggestion
          description: ID of the audience from which to fetch profiles.
```

File: customer_data_platform/authorized_api/customer_data_platform_log_api.yaml [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+          schema:
+            type: string
+            format: uuid
+            example: ca5b925e-6888-46fe-a0bc-55907c5af050
+          required: true
+        - in: path
+          name: audience_version
+          description: An audience version for which profiles belong
```
Comment (prk3) [inline customer_data_platform/authorized_api/customer_data_platform_log_api.yaml:74]:
```suggestion
          description: Version of the audience. This should be the version reported by audiences API (link).
```

File: customer_data_platform/authorized_api/customer_data_platform_log_api.yaml [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+          schema:
+            type: integer
+            minimum: 0
+            example: 2
+          required: true
+        - in: query
+          name: limit
+          description: How many last profiles should be returned
```
Comment (prk3) [inline customer_data_platform/authorized_api/customer_data_platform_log_api.yaml:82]:
```suggestion
          description: How many last profiles to return
```

File: customer_data_platform/authorized_api/customer_data_platform_log_api.yaml [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+          schema:
+            type: integer
+            maximum: 1000
+            minimum: 1
+            example: 10
```
Comment (prk3) [inline customer_data_platform/authorized_api/customer_data_platform_log_api.yaml:87]: Can we say that the default is 100? Same for offest, it should be 0 by default.

File: customer_data_platform/authorized_api/customer_data_platform_log_api.yaml [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+          type: string
+          format: date-time
+          example: "2022-10-17T09:46:16.343024Z"
+        last_activity_at:
+          description: Last time the profile met audience conditions
+          type: string
+          format: date-time
+          example: "2022-10-17T09:49:16.123456Z"
+        is_user:
+          description: Whether the profile can be identified as a user or is just a visitor
```
Comment (prk3) [inline customer_data_platform/authorized_api/customer_data_platform_log_api.yaml:146]:
```suggestion
          description: Whether the profile can be identified as a user (has user_id) or a visitor (does not have user_id).
```

**Review by prk3 — CHANGES_REQUESTED** [review]

File: customer_data_platform/authorized_api/customer_data_platform_log_api.yaml (hunk 5 above)
Comment (havaker) [reply customer_data_platform/authorized_api/customer_data_platform_log_api.yaml:87]: Sure!

**Review by kororokke — APPROVED** [review]

**Comment by havaker** [comment]
Added default values requested by @prk3:
image

File: customer_data_platform/authorized_api/customer_data_platform_log_api.yaml [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+          name: audience_id
+          description: ID of the audience from which to fetch profiles.
+          schema:
+            type: string
+            format: uuid
+            example: ca5b925e-6888-46fe-a0bc-55907c5af050
+          required: true
+        - in: path
+          name: audience_version
+          description: Version of the audience. This should be the version reported by audiences API (link).
```
Comment (prk3) [inline customer_data_platform/authorized_api/customer_data_platform_log_api.yaml:74]: Please replace `(link)` with a real link to GET /audience/id endpoint. :smiley:

Comment (havaker) [reply customer_data_platform/authorized_api/customer_data_platform_log_api.yaml:74]: I'd rather delete it, inserting links to to a path produced by our webui generation machinery feels meeh and decreases maintainability.

Marking as resolved, got a green light in out-of-band direct conversation.

**Pull request merged**